src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -200,12 +200,27 @@ private async Task<Document> InlineTemporaryAsync(Document document, VariableDec
 
             var topmostParentingExpressions = nonConflictingIdentifierNodes
                 .Select(ident => GetTopMostParentingExpression(ident))
-                .Distinct();
+                .Distinct().ToList();
 
             var originalInitializerSymbolInfo = semanticModel.GetSymbolInfo(variableDeclarator.Initializer.Value, cancellationToken);
 
             // Make each topmost parenting statement or Equals Clause Expressions semantically explicit.
-            updatedDocument = await updatedDocument.ReplaceNodesAsync(topmostParentingExpressions, (o, n) => Simplifier.Expand(n, semanticModel, workspace, cancellationToken: cancellationToken), cancellationToken).ConfigureAwait(false);
+            updatedDocument = await updatedDocument.ReplaceNodesAsync(topmostParentingExpressions, (o, n) =>
+            {
+                var node = Simplifier.Expand(n, semanticModel, workspace, cancellationToken: cancellationToken);
+
+                // If we are inlining an expression into a single location and that single location is inside a conditional method call,
+                // then it is likely that the number of times the expression is called matters,
+                // and that the user hasn't realised that the expression won't be called unless the correct conditional symbol is defined.
```

Comment:
not certain it's relevant how many times it is inlined.  i think just having the warning in each case (specifically becaues that code may not run at all) is totally reasonable.

Reply:
The behavior here does seem surprising to me: if all of the locations are conditional, it still seems we'd want to warn, right? I get that sticking a warning on all of the uses might be irritating but it seems better to over-warn than be silent here.

Thoughts?